### DISCUSSION TITLE
Start two concurrent Sumatra instances; with normal User- AND with Administrative-Rights?

### DISCUSSION THREAD

**Discussion by esrat** [sumatrapdfreader/sumatrapdf#2346] [Q&A]
Hi,
I do like SumatraPDF a lot, because it's quick and works tidy / portable!

But there is an issue, caused by Windows, which hurts my user experience quite much:
Obviously Windows does not allow to open files from a process **without administrative rights** into an already running process **with administrative rights**.
For example: My file manager mostly runs with administrative rights, to be able to access most directories. If no Sumatra instance exists, opening the first PDF file will spawn one which will inherit the administrative rights. If I need the document later on again, I will let the Sumatra window open. In the meantime I could use some other application without administrative rights, maybe the web browser. If I choose to open a PDF file from a web page into Sumatra - this will do nothing (at least in older versions)!
The web browser still downloads the file to some temporary path but it cannot advise the running Sumatra instance to open it. If the running instance would not run in administrative mode (or would not exist at all), this would succeed normally.

Maybe I did not need to write that much. - I just realized that the most current version 3.3.3 does already start another instance, just to show a message "SumatraPDF is running as admin and cannot open files from a non-admin process". So the issue is known already.

My old intention for this thread was to ask about the option **ReuseInstance** which seems to do nothing. I wanted to ask how it should be possible to open multiple instances - explicitly one with and another one without administrative rights!
But the most current version does already do this - but ignores the given file path completely although it is configured with the default setting **ReuseInstance = false**!?

So why does this happen? Why does the newly created process only show this message instead of showing the document???

I wouldn't have any problem with actively choosing the window to open the new document in. - Other tools do so in a perfectly logical way.
For example: The editor Notepad++ does allow opening multiple instances (e.g. with the command line parameters "-multiInst -nosession") - and whenever the user chooses to open another document within Notepad++, the window which was focused last will be used. This would be a perfect solution for the admin-non-admin-problem of Sumatra and multiple instances in general!
I would be perfectly fine with managing two Sumatra windows, bringing the admin one to the front when opening something with my file manager and bringing the other one the front when opening something with the browser.

Another hint: I also tried to start both instances manually (of course with NO ReuseInstance set!). This does not work either. If the non-admin instance is started first, the admin one will just bring the first one up to the front (not opening any new instance). And if the admin instance is started first, the non-admin one does show the same message as quoted above - even so no file path (or any other parameter) was given!?

Is there any way to achieve the goal of two instances with and without administrative rights? Or open multiple instances at all?

**Comment by kjk** [comment]
When you start non-admin Sumatra and then double-click on a PDF file from your admin file manager, does it open the file?

I haven't tested it yet, but if it does, that's the solution: keep non-admin Sumatra process running.

Running admin and non-admin process would not be a reliable fix.

**Comment by esrat** [comment]
Sorry for my late answer.

Yes, this works but it is not convenient.
To use this workaround, I would have to start a SumatraPDF instance prematurely and I would have to keep it running even though I do not need it (any more)!
This would contradict the efficient usage of the program altogether!

Like I stated before, I do like the portability of Sumatra and its efficient usability (hitting [Esc] to close, re-opening within a split second, …). I do not even have nor need a start menu entry, because I don’t need to start it explicitly without a document to open.

I think, I found **a more convenient workaround** myself:
I can simply copy the Sumatra directory and use these copies independent from each other!
That means, the copy which was originally registered for Windows will stay there. It’s much more likely that I will open a PDF file from my file manager, so that copy will most likely run with administrative rights and it will be re-used for every other file I open from my file manager.
For downloads in my browser I chose to open the file within the other copy of the program which now works, because it has a different path to the executable. From now on, my browser opens every temporary PDF file with this second copy by default – and I do have what I wanted in the first place.
I can open files from disk as well as from web pages now and they will be shown in the corresponding window.

But this works only for applications like the browser which do accept a user defined program to open PDF files with.

**Reply by GitHubRulesOK** [reply]
I as a user, run SumatraPDF in admin class on web files, generally without concern since at present SumatraPDF cannot run scripted pdf files thus 99.x% safe for calling web based pdf (NOTE:- the complaint is it will not run [breakout.pdf](https://github.com/osnr/horrifying-pdf-experiments) game or change the date or change the appearance or goto the authors designated page action, nor phone home when a page is opened.) So IF I thought a browser session was likely to cause me to open a viral PDF (which HAS to download through virus checker) I would simply open it in the windows sandboxed browser or using the SumatraPDF Pale Moon/Waterfox classic Browser plugin (which is not recommended anymore, and not sure what happens when PaleMoon next update the FF plugins soon in v30)

**Reply by esrat** [reply]
To open web PDF files in administrative mode I would have to run the calling application in administrative mode too. That means, I would have to run my browser in administrative mode!
I do not consider this a good idea. The browser is the common gateway to the Internet and so it is considered a persistent security risk! Even the developers would not support such thing, at least not in my case: I am using a Firefox ESR portable. Even when I do start the FirefoxPortable.exe from my administrative file manager, so it also gets administrative rights, but the resulting firefox.exe-processes are being downgraded again! That behaviour (I guess, for security reasons) was causing my original problem, because I cannot load temporary downloaded files through initiative of these browser processes (without administrative rights).

**Reply by kfurter-sumatra** [reply]
Using 2 different installations of Sumatra was a great idea. Using my favorite filemanager I have configured Sumatra as viewer using the portable version in the filemanager directory. And for anything else (like viewing in Thunderbird) the normal installed version.
Thanks for the idea!

**Comment by Ander3232** [comment]
It's silly. If I have a PDF open in Sumatra, then try to open another via a Start Menu shortcut (I keep shortcuts there for many frequently used documents), I get that error message—even though I'm logged in as an Administrator, so what's the difference? I'd set the shortcuts to "Run as Administrator"—but for some reason, that option is disabled in their Properties. (Guess it'd be too easy otherwise, LOL)

I've used many PDF viewers, and Sumatra's the first that's acted this way. I don't know how all the others got around the "problem", but it'd be great if Sumatra's coders would at least make it an option for those of us who aren't concerned about it.

**Comment by GitHubRulesOK** [comment]
@Ander3232 
there may be some difference in your system configuration
1) I am admin user most of the time
2) I use many sessions (many 32 and 64 bit installed and portable versions mixed) without ever seeing that message
3) start or taskbar or 32/64 popup launcher or "send to" or shortcuts or from other apps none of those cause the message
![image](https://user-images.githubusercontent.com/12754913/199872600-9034542d-d302-4a45-b8a0-bd8fff1a8140.png)
